subprojects/workers/src/main/java/org/gradle/workers/internal/IsolatedClassLoaderUtil.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal;
+
+import org.gradle.initialization.GradleApiUtil;
+import org.gradle.internal.classloader.FilteringClassLoader;
+import org.gradle.internal.classloader.VisitableURLClassLoader;
+import org.gradle.internal.classpath.DefaultClassPath;
+import org.gradle.internal.reflect.DirectInstantiator;
+
+import java.io.File;
+
+public class IsolatedClassLoaderUtil {
+    static FilteringClassLoader.Spec gradleApiSpec;
+
+    private static FilteringClassLoader.Spec getGradleApiSpec() {
+        if (gradleApiSpec == null) {
+            gradleApiSpec = GradleApiUtil.apiSpecFor(IsolatedClassLoaderUtil.class.getClassLoader(), DirectInstantiator.INSTANCE);
```

Comment:
I think that `ClassLoaderRegistry` should be involved in this decision, as this is the thing that is responsible for defining the implementation + API filter.